adam-core/src/main/scala/org/bdgenomics/adam/converters/SAMRecordConverter.scala

```diff
@@ -17,168 +17,187 @@
+          builder.setOldPosition(samRecord.getIntegerAttribute("OP").toLong - 1)
+          builder.setOldCigar(samRecord.getStringAttribute("OC"))
         }
-      builder.setEnd(end)
-      // set mapping quality
-      val mapq: Int = samRecord.getMappingQuality
 
-      if (mapq != SAMRecord.UNKNOWN_MAPPING_QUALITY) {
-        builder.setMapq(mapq)
-      }
+        val end = samRecord.getCigar.getCigarElements
+          .asScala
+          .filter((p: CigarElement) => p.getOperator.consumesReferenceBases())
+          .foldLeft(start - 1) {
+            (pos, cigarEl) => pos + cigarEl.getLength
+          }
+        builder.setEnd(end)
+        // set mapping quality
+        val mapq: Int = samRecord.getMappingQuality
 
-      // set mapping flags
-      // oddly enough, it appears that reads can show up with mapping info (mapq, cigar, position)
-      // even if the read unmapped flag is set...
-      if (samRecord.getReadUnmappedFlag) {
-        builder.setReadMapped(false)
-      } else {
-        builder.setReadMapped(true)
-        if (samRecord.getReadNegativeStrandFlag) {
-          builder.setReadNegativeStrand(true)
+        if (mapq != SAMRecord.UNKNOWN_MAPPING_QUALITY) {
+          builder.setMapq(mapq)
         }
-        if (!samRecord.getNotPrimaryAlignmentFlag) {
-          builder.setPrimaryAlignment(true)
+
+        // set mapping flags
+        // oddly enough, it appears that reads can show up with mapping info (mapq, cigar, position)
+        // even if the read unmapped flag is set...
+        if (samRecord.getReadUnmappedFlag) {
+          builder.setReadMapped(false)
         } else {
-          // if the read is not a primary alignment, it can be either secondary or supplementary
-          // - secondary: not the best linear alignment
-          // - supplementary: part of a chimeric alignment
-          builder.setSupplementaryAlignment(samRecord.getSupplementaryAlignmentFlag)
-          builder.setSecondaryAlignment(!samRecord.getSupplementaryAlignmentFlag)
+          builder.setReadMapped(true)
+          if (samRecord.getReadNegativeStrandFlag) {
+            builder.setReadNegativeStrand(true)
+          }
+          if (!samRecord.getNotPrimaryAlignmentFlag) {
+            builder.setPrimaryAlignment(true)
+          } else {
+            // if the read is not a primary alignment, it can be either secondary or supplementary
+            // - secondary: not the best linear alignment
+            // - supplementary: part of a chimeric alignment
+            builder.setSupplementaryAlignment(samRecord.getSupplementaryAlignmentFlag)
+            builder.setSecondaryAlignment(!samRecord.getSupplementaryAlignmentFlag)
+          }
         }
       }
-    }
 
-    // Position of the mate/next segment
-    val mateReference: Int = samRecord.getMateReferenceIndex
+      // Position of the mate/next segment
+      val mateReference: Int = samRecord.getMateReferenceIndex
```

Comment:
Is this indentation intentional? I'm surprised that Scalariform didn't complain.